control/src/lib.rs

```diff
@@ -356,82 +292,120 @@ pub mod pallet {
 		/// - `name`: Org name.
 		/// - `cid`: IPFS content identifier.
 		/// - `org_type`: Individual | Company | Dao | Hybrid.
-		/// - `access`: Open (anyone can join) | Voting (membership voting) |
-		/// 	Controller (controller invites).
-		/// - `fee_model`: NoFees | Reserve (amount reserved in user account) |
-		/// 	Transfer (amount transfered to Org treasury).
-		/// - `fee`: fees amount to be applied to new members based on fee model (in Protocol tokens).
+		/// - `access_model`:
+		/// - `fee_model`:
+		/// 
+		/// Optional parameters:
+		/// - `member_limit`: max members. Default: MaxMembers.
+		/// - `member_fee`: fees amount to be applied to new members based on fee model (in `gov_asset` tokens).
 		/// - `gov_asset`: control assets to empower actors.
 		/// - `pay_asset`: asset used for payments.
-		/// - `member_limit`: max members, if 0 == no limit.
 		/// - `deposit`: initial deposit for the org treasury (in Protocol tokens).
 		///
 		/// Emits `OrgCreated` event when successful.
 		///
 		/// Weight: `O(1)`
 		#[pallet::weight(T::WeightInfo::create_org())]
+		// Tests are not working without this deprecated transactional macro
+		#[transactional]
 		pub fn create_org(
 			origin: OriginFor<T>,
-			controller_id: T::AccountId,
-			name: BoundedVec<u8, T::StringLimit>,
-			cid: BoundedVec<u8, T::StringLimit>,
+			name: String<T>,
+			cid: String<T>,
 			org_type: OrgType,
-			access: AccessModel,
+			access_model: AccessModel,
 			fee_model: FeeModel,
-			fee: T::Balance,
-			gov_asset: T::CurrencyId,
-			pay_asset: T::CurrencyId,
-			member_limit: u64,
+			// Optional parameters:
+			member_limit: Option<MemberLimit>,
+			membership_fee: Option<T::Balance>,
+			gov_currency: Option<T::CurrencyId>,
+			pay_currency: Option<T::CurrencyId>,
 			deposit: Option<T::Balance>,
-			// mint: T::Balance,
-			// burn: T::Balance,
-			// strategy: u16,
 		) -> DispatchResult {
 			let sender = ensure_signed(origin.clone())?;
 
-			let mut org_deposit: T::Balance = T::MinimumDeposit::get();
-			if deposit.is_some() {
-				org_deposit = deposit.unwrap();
-				ensure!(org_deposit >= T::MinimumDeposit::get(), Error::<T>::MinimumDepositTooLow);
-			}
+			// Provide default values for optional parameters:
+			let member_limit = member_limit.unwrap_or(T::MaxMembers::get());
+			let gov_currency = gov_currency.unwrap_or(T::ProtocolTokenId::get());
+			let pay_currency = pay_currency.unwrap_or(T::PaymentTokenId::get());
+			let deposit = deposit.unwrap_or(T::MinimumDeposit::get());
 
-			let creator_balance = T::Currency::free_balance(T::ProtocolTokenId::get(), &sender);
-			ensure!(creator_balance >= org_deposit, Error::<T>::BalanceTooLow);
+			ensure!(
+				org_type == OrgType::Individual || org_type == OrgType::Dao,
+				Error::<T>::WrongOrganizationType
+			);
+			ensure!(deposit >= T::MinimumDeposit::get(), Error::<T>::MinimumDepositTooLow);
+			if fee_model != FeeModel::NoFees && membership_fee.is_none() {
```

Comment:
I'd suggest to make all validation of the same type.
So instead of `if`, use `ensure!` macro the same way as all checks before this one:
```
ensure!(
    fee_model == FeeModel::NoFees || membership_fee.is_some(),
    Error::<T>::MissingParameter
);
```